src/usr/share/torwatchdog/torwatchdog.py

```diff
@@ -15,174 +15,226 @@
+        raise InitializationException(
+            'File %s must be owned by root.' % CONFIGURATION_PATHNAME)
+    if bool(config_file_stat.st_mode & (stat.S_IROTH | stat.S_IWOTH | stat.S_IXOTH)):
+        raise InitializationException(
+            "File %s cannot have 'other user' access permissions set."
+            % CONFIGURATION_PATHNAME)
 
-    logger.info('Creating directory %s.' % os.path.join(system_path, program_dirs))
 
+def create_directory(system_path, program_dirs, uid, gid, mode):
+    """Creates directories if they do not exist and sets the specified ownership and
+    permissions.
+
+    system_path: The system path that the directories should be created under. These are
+      assumed to already exist. The ownership and permissions on these directories are not
+      modified.
+    program_dirs: A string representing additional directories that should be created under
+      the system path that should take on the following ownership and permissions.
+    uid: The system user ID that should own the directory.
+    gid: The system group ID that should be associated with the directory.
+    mode: The umask of the directory access permissions.
+    """
+    logger.info('Creating directory %s.', os.path.join(system_path, program_dirs))
+
+    path = system_path
     for directory in program_dirs.strip('/').split('/'):
-        path = os.path.join(system_path, directory)
+        path = os.path.join(path, directory)
         if not os.path.isdir(path):
             # Will throw exception if file cannot be created.
             os.makedirs(path, mode)
         os.chown(path, uid, gid)
         os.chmod(path, mode)
 
 
-# Drops escalated permissions forever to the specified user and group.
-#
-# uid: The system user ID to drop to.
-# gid: The system group ID to drop to.
 def drop_permissions_forever(uid, gid):
-    logger.info('Dropping permissions for user %s.' % process_username)
-    os.initgroups(process_username, gid)
+    """Drops escalated permissions forever to the specified user and group.
+
+    uid: The system user ID to drop to.
+    gid: The system group ID to drop to.
+    """
+    logger.info('Dropping permissions for user %s.', PROCESS_USERNAME)
+    os.initgroups(PROCESS_USERNAME, gid)
     os.setgid(gid)
     os.setuid(uid)
 
 
-# Configures the tor proxy settings.
-#
-# config: The program configuration object, mostly based on the configuration file.
 def configure_tor_proxy(config):
+    """Configures the tor proxy settings.
+
+    config: The program configuration object, mostly based on the configuration file.
+    """
     # Set socks proxy and wrap the urllib module
-    # TODO: Consider choosing a randomly available TCP port.
-    socks.setdefaultproxy(socks.PROXY_TYPE_SOCKS5, '127.0.0.1', int(config['tor_socks_port']))
+    # TODO: Eventually consider choosing a randomly available TCP port.
+    socks.setdefaultproxy(
+        socks.PROXY_TYPE_SOCKS5, '127.0.0.1', int(config['tor_socks_port']))
```

Comment:
`config['tor_socks_port']` is already definitely an int, no need to typecast it again.